endpoints/openrtb2/auction.go

```diff
@@ -1980,6 +2007,73 @@ func getAccountID(pub *openrtb2.Publisher) string {
 	return metrics.PublisherUnknown
 }
 
+// Returns the account from the request
+func getAccountIdFromRawRequest(hasStoredRequest bool, storedRequest json.RawMessage, originalRequest []byte) (string, bool, []error) {
+	var accountId string
+	var isAppReq bool
+	var err error
+
+	request := originalRequest
+	if hasStoredRequest {
+		request = storedRequest
+	}
+
+	accountId, isAppReq, err = searchAccountId(request)
+	if err != nil {
+		return "", isAppReq, []error{err}
+	}
+
+	// In case the stored request did not have account data we specifically search it in the original request
+	if accountId == "" && hasStoredRequest {
+		accountId, _, err = searchAccountId(originalRequest)
+		if err != nil {
+			return "", isAppReq, []error{err}
+		}
+	}
+
+	if accountId == "" {
+		return metrics.PublisherUnknown, isAppReq, nil
+	}
+
+	return accountId, isAppReq, nil
+}
+
+func searchAccountId(request []byte) (string, bool, error) {
+	queries := [...]struct {
+		isApp bool
+		key   []string
+	}{
+		{true, []string{"app", "publisher", "ext", openrtb_ext.PrebidExtKey, "parentAccount"}},
+		{true, []string{"app", "publisher", "id"}},
+		{false, []string{"site", "publisher", "ext", openrtb_ext.PrebidExtKey, "parentAccount"}},
+		{false, []string{"site", "publisher", "id"}},
+	}
```

Comment:
I like this approach. Please define the `queries` variable at the package level (up one level) to avoid allocating memory for every request. Consider renaming to something like "accountIdSearchPath" to make the relationship with this method clear.